### PR TITLE
LPE firmware creation for px4fmu-v2 and param default overrides

### PR DESCRIPTION
**Pull request by jgoppert**
This adds LPE firmware build for px4fmu-v2 and also add a mechanism for param default overrides. Right now a lot of users are struggling to build LPE since it is no-longer default and I'm trying to make it easier for them. It would be great if it could be an option for QGC @dagar @LorenzMeier .

The param default override  is used to make sure the user starts up with LPE as default. I made it a json dict string so it is very easy to add to other configs.

```cmake
set(PARAM_DEFAULT_OVERRIDES "{\\\"SYS_MC_EST_GROUP\\\": 1}")
```

### CODE REVIEW

**Comment by dagar** [comment]
There isn't really a good mechanism for selecting alternate firmware versions for the same branch (and board). Any ideas @DonLakeFlyer?

Any realistic options for squeezing it in? Why do we need mpu9250 in px4fmu-v2_default?

**Comment by dagar** [comment]
We probably don't want to go here, but could we use the 2nd MB for anything? I'm thinking of the px4io binary.

**Comment by jgoppert** [comment]
@LorenzMeier any harm in merging this? We can figure out how to distribute LPE as a separate issue, but I think having the make px4fmu-v2_lpe target working is a big improvement for users. I can remove it form the check targets if that would make sense. @dagar.

File: Makefile [hunk 1]
```diff
@@ -207,6 +207,7 @@ qgc_firmware: \
 	check_mindpx-v2_default \
 	check_px4fmu-v1_default \
 	check_px4fmu-v2_default \
+	check_px4fmu-v2_lpe \
```
Comment (jgoppert) [inline Makefile]: @dagar What is the impact of leaving this here?

Comment (dagar) [inline Makefile]: Everything under `qgc_firmware` is built on travis-ci and pushed to s3 for master/beta/stable.

Comment (jgoppert) [inline Makefile]: OK cool, so that should mean it should be accessible for QGC if we give it the correct url?

Comment (dagar) [inline Makefile]: Actually, I bet it's skipped in the deploy with the naming. Checking...

Comment (dagar) [inline Makefile]: Needs something here as well for the upload - https://github.com/PX4/Firmware/blob/master/.travis.yml#L82

I don't know how that line got so convoluted, but it needs to copy the firmware to s3deploy-branch/ renamed as px4fmu-v2_lpe.px4

If you expand the travis-ci output near the bottom you'll see it running.

Comment (jgoppert) [inline Makefile]: Yikes, any issue with me cleaning up packaging using cmake?

Comment (jgoppert) [inline Makefile]: Working now on tweaking it to match your layout and filenames.

**Comment by jgoppert** [comment]
@dagar @LorenzMeier I think this should work with s3 now and it moves all of the makefile logic to python so it is easier to read.

**Comment by dagar** [comment]
That's a lot of code just to copy some files :wink: 
A lot of this exists because of limitations with the travis-ci s3 deploy which only works per directory. This is a good excuse to pull more of the s3 upload into our tools and lessen our dependence on travis-ci (which is becoming unbearably slow). Sorry, should have just done that earlier.

Do we need the nuttx cpack for anything? It's just the px4 file zipped? Can it be removed from ALL and not run by default?

**Comment by jgoppert** [comment]
Agreed it is a bit verbose. Cpack is the cmake way to build packages for distribution.  If you can take the cpack packages directly into s3 90% of thhe code will disappear.  But I think for now as an intermediate step the code is reasonable and pretty straightforward.  More than the makefile atleast.

**Comment by jgoppert** [comment]
Note sure what you mean about make px4fmu-v2_default package target being in ALL? I don't believe it is.

**Comment by dagar** [comment]
Isn't the build on travis-ci right now creating the package without calling the package target? Meaning it's in ALL.

I guess I'm not seeing the point of creating these packages for nuttx builds. A px4 file is already the binary (compressed) with airframe and parameter metadata. Now we're zipping that plus another copy of the metadata? What's the benefit?

I was just going to pull the s3 upload into a simple script and stop relying on the travis-ci functionality. I suppose this could just be a custom target in cmake.
https://github.com/PX4/Firmware/blob/4f0ebac3f3f85bd3794d6e0d6e2cb3ead1e0af8f/Tools/s3put.sh

**Comment by jgoppert** [comment]
@dagar and I talked offline, but just to clarify the packaging target is being called here: https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2FPX4%2FFirmware%2Fpull%2F6307%2Ffiles%23diff-b67911656ef5d18c4ae36cb6741b7965R263&sa=D&sntz=1&usg=AFQjCNFOckkcc6Z9bjrFhkT0qlY5-eLykw

So it will only run for what we want to package for s3 upload.

**Comment by jgoppert** [comment]
OK, thanks to some help from @dagar I've got it uploaded to s3 now: https://travis-ci.org/PX4/Firmware/jobs/191551546#L922

I think this is ready for a final look over, working for me here @dagar @LorenzMeier.

**Pull request merged**